app/subcommands/stop_subcommand.py

```diff
@@ -62,23 +62,37 @@ def _kill_pid_in_file_if_exists(self, pid_file_path):
 
         if not matched_proc_command:
             self._logger.info(
-                "PID {0} is running, but command '{1}' is not a clusterrunner command".format(pid, proc_command))
+                "PID {0} is running, but command '{1}' is not a clusterrunner command".format(proc.pid, proc_command))
             return
 
         # Try killing gracefully with SIGTERM first. Then give process some time to gracefully shutdown. If it
         # doesn't, perform a SIGKILL.
         # @TODO: use util.timeout functionality once it gets merged
-        os.kill(int(pid), signal.SIGTERM)
+        self.kill_proc_and_children(proc, signal.SIGTERM)
         sigterm_start = time.time()
 
         while (time.time()-sigterm_start) <= self._sigterm_sigkill_grace_period_sec:
-            if not psutil.pid_exists(int(pid)):
+            if not proc.is_running():
```

Comment:
What if the parent process dies in response to SIGTERM but the child processes don't? It seems like this logic would not send a SIGKILL to the children in that case.